R/add_overall_ard.R

```diff
@@ -0,0 +1,87 @@
+#' ARD add overall column
+#'
+#' Adds a column with overall summary statistics to tables
+#' created by `tbl_ard_summary()`.
+#'
+#' @param x (`tbl_ard_summary`)\cr
+#'   A stratified 'gtsummary' table
+#' @param cards (`card`)\cr
+#'   An ARD object of class `"card"` typically created with `cards::ard_*()` functions.
+#' @param last (scalar `logical`)\cr
+#'   Logical indicator to display overall column last in table.
+#'   Default is `FALSE`, which will display overall column first.
+#' @param col_label (`string`)\cr
+#'   String indicating the column label. Default is `"**Overall**"`
+#' @param statistic ([`formula-list-selector`][syntax])\cr
+#'   Override the statistic argument in initial `tbl_*` function
+#'   call. Default is `NULL`.
+#' @param digits ([`formula-list-selector`][syntax])\cr
+#'   Override the digits argument in initial `tbl_*` function
+#'   call. Default is `NULL`.
+#' @inheritParams rlang::args_dots_empty
+#'
+#' @author Daniel D. Sjoberg
+#' @name add_overall_ard
+#' @return A `gtsummary` of same class as `x`
+#' @examples
+#' # Example 1 ----------------------------------
+#' # build primary table
+#' tbl <-
+#'   cards::ard_stack(
+#'     trial,
+#'     .by = trt,
+#'     cards::ard_continuous(variables = age),
+#'     cards::ard_categorical(variables = grade),
+#'     .missing = TRUE,
+#'     .attributes = TRUE,
+#'     .total_n = TRUE
+#'   ) |>
+#'   tbl_ard_summary(by = trt)
+#'
+#' # create ARD with overall results
+#' ard_overall <-
+#'   cards::ard_stack(
+#'     trial,
+#'     cards::ard_continuous(variables = age),
+#'     cards::ard_categorical(variables = grade),
+#'     .missing = TRUE,
+#'     .attributes = TRUE,
+#'     .total_n = TRUE
+#'   )
+#'
+#' # add an overall column
+#' tbl |>
+#'   add_overall(cards = ard_overall)
+NULL
+
+#' @rdname add_overall_ard
```

Comment:
Why is it on a different page even if it is a S3? I guess to avoid having too many entries? if it is so it is fine for me

Reply:
Since 99.9% of gtsummary users don't care about ARDs and the ARD functions have an additional argument not used by the other `add_overall()` methods, I thought it can live in it's own space. Over time, we'll be adding other `tbl_ard_*` `add_overall()` methods.